packages/components/src/RichTooltip/RichLayout/RichLayout.component.js

```diff
@@ -37,21 +37,32 @@ const Footer = ({ id, children, className }) => (
 );
 Footer.propTypes = TooltipPropTypes;
 
-export default function RichLayout(props) {
-	return (
-		<span id={props.id}>
-			<Header id={`${props.id}-header`}>
-				{Inject.getReactElement(props.getComponent, props.Header)}
-			</Header>
-			<Body id={`${props.id}-body`} className={props.className}>
-				{props.text && <p>{props.text}</p>}
-				{!props.text && Inject.getReactElement(props.getComponent, props.Content)}
-			</Body>
-			<Footer id={`${props.id}-footer`}>
-				{Inject.getReactElement(props.getComponent, props.Footer)}
-			</Footer>
-		</span>
-	);
+export default class RichLayout extends React.Component {
+	constructor(props) {
+		super(props);
+		this.richLayout = React.createRef();
+	}
+
+	focusRichLayout() {
+		this.richLayout.current.focus();
+	}
+
+	render() {
+		return (
+			<span id={this.props.id} ref={this.richLayout} tabIndex="-1">
```

Comment:
It makes sense here to use FocusTrap I guess

Reply:
And do you know why a span was used here? I'm wondering?

Reply:
Legacy I think to wrap the element.

Reply:
Could you profit of your PR to change it to a more convenient div, please?

Reply:
done